src/plugins/data_views/common/fields/data_view_field.ts

```diff
@@ -41,16 +46,25 @@ export class DataViewField implements DataViewFieldBase {
    * Kbn field type, used mainly for formattering.
    */
   private readonly kbnFieldType: KbnFieldType;
+  /**
+   * Returns context configuration for the actual field, e.g if it's part of an ECS fields list
+   * @private
+   */
+  private readonly getContext: (() => DataViewFieldContext) | undefined;
 
   /**
    * DataView constructor
    * @constructor
    * @param spec Configuration for the field
+   * @param getContext Get Context of the field
    */
-  constructor(spec: FieldSpec) {
+  constructor(spec: FieldSpec, getContext?: (() => DataViewFieldContext) | undefined) {
```

Comment:
@mattkime wdyt about that, rather than passing in a instance of the FieldList, an optional function returning info about the context is passed in, it could also be a function returning `true` or `false` when ECS in the given context is available. This makes is flexible and useable with DataView and Lens (which uses DataViews in a custom way), and if needed it can also be adapted for DataViewLazy (when needed)